control-plane/csi-driver/src/filesystem/mod.rs

```diff
@@ -0,0 +1,57 @@
+/// Filesystem types and definition.
+pub mod types;
+
+/// Filesystem freeze low level code.
+pub mod fsfreeze;
+
+use std::time::Duration;
+use tokio::{process::Command, time::timeout};
+use tonic::Status;
+use tracing::debug;
+
+// Name of the csi-node binary to be used to make a call.
+const CSI_NODE_BINARY: &str = "csi-node";
+// Name of the argument for the `csi-node fsfreeze/fsunfreeze` subcommand.
+const FREEZE_VOLUME_ID_ARG_NAME: &str = "--volume-id";
+
+/// Creates a child process that calls up the csi-node binary to execute FIFREEZE and FITHAW using
+/// the volume id.
+pub async fn fsfreeze_with_timeout(
+    volume_id: &str,
+    freeze_op: fsfreeze::FsFreezeOpt,
+    timeout_duration: Duration,
+) -> Result<(), Status> {
+    debug!(volume_id=volume_id, timeout=?timeout_duration,"{freeze_op}");
+    let mut command = Command::new(CSI_NODE_BINARY)
+        .arg(freeze_op.as_ref())
+        .arg(FREEZE_VOLUME_ID_ARG_NAME)
+        .arg(volume_id)
+        .spawn()
+        .map_err(|error| {
+            Status::failed_precondition(format!(
+                "Failed to spawn child process for {freeze_op} for {volume_id}, {error}"
+            ))
+        })?;
+
+    let execute_with_timeout_result = timeout(timeout_duration, command.wait()).await;
```

Comment:
Shouldn't the timeout be set only at the calling side, and allow the future to run even after the timeout?
Otherwise we'll keep stacking freeze ioctl?